hail/python/hail/methods/statgen.py

```diff
@@ -1989,13 +1989,15 @@ def realized_relationship_matrix(call_expr) -> BlockMatrix:
                         __n_called=agg.count_where(hl.is_defined(mt.__gt)))
     mt = mt.select_rows(__mean_gt=mt.__AC / mt.__n_called,
                         __centered_length=hl.sqrt(mt.__ACsq - (mt.__AC ** 2) / mt.__n_called))
-    mt = mt.filter_rows(mt.__centered_length > 0.1)  # truly non-zero values are at least sqrt(0.5)
+    fmt = mt.filter_rows(mt.__centered_length > 0.1)  # truly non-zero values are at least sqrt(0.5)
 
-    normalized_gt = hl.or_else((mt.__gt - mt.__mean_gt) / mt.__centered_length, 0.0)
-    bm = BlockMatrix.from_entry_expr(normalized_gt)
-
-    return (bm.T @ bm) / (bm.n_rows / bm.n_cols)
+    normalized_gt = hl.or_else((fmt.__gt - fmt.__mean_gt) / fmt.__centered_length, 0.0)
 
+    try:
+        bm = BlockMatrix.from_entry_expr(normalized_gt)
+        return (bm.T @ bm) / (bm.n_rows / bm.n_cols)
+    except FatalError as fe:
+        raise FatalError("Could not convert MatrixTable to BlockMatrix. It's possible all variants were dropped by variance filter.") from fe
```

Comment:
In other user-focused error messages we've provided suggested fixes. Is it appropriate to do that here?

```
FatalError: Could not convert MatrixTable to BlockMatrix. It's possible all variants were dropped by variance filter.
Check that the input MatrixTable has at least two samples in it:  mt.count_cols().
Check that the input MatrixTable has at least one non-zero-variance variant: ...
```
Hmm. It'd be nice if the variance computation above was also in a user accessible method so we could just include it there. Maybe that's a step too far.

Reply:
I'll do the check at least two samples thing. I don't know how to tell them to calculate the variance. Also, this error should virtually never happen.